### DISCUSSION TITLE
É possível aprender React apenas com a documentação?

### DISCUSSION THREAD

**Discussion by sou-gabriel** [frontendbr/forum#2086] [Perguntas e Respostas]
Galera consegui obter uma excelente base de JavaScript por meio do [Curso de JavaScript Roger Melo](https://rogermelo.com.br), mas me deparo com a seguinte situação: preciso aprender um framework e estou na dúvida se realmente vale a pena investir uma grana em um bom curso ou tento aprender apenas pela documentação (já que a base já conquistei). O que vocês poderiam me recomendar galera?

**Comment by ninetails** [comment]
Varia de pessoa pra pessoa. Apesar da documentação do React ser rica e bem feita, pode ter coisa que pode estar desatualizado.

E não recomendo tentar aprender React somente por lá pois o React em si é só uma biblioteca de _view_. Na documentação falará somente dela, e o que vale mesmo para React é aprender todo o ecossistema que o envolve: como estilizar, _stores_, meta frameworks que vc pode utilizar como Next e Gatsby...

Estudando somente ele puro, creio que não irá ter progresso além de fazer um <abbr title="Single Page Application">SPA</abbr>.

Tem curso de todos os gostos e de todos os formatos a todos os preços (claro que essas variáveis vão influenciar na qualidade). Também pode ser que dê azar e pegue tutoriais e cursos ultrapassados, mas ainda sim creio que pra começar seria muito mais prático por eles.

Recomendo sim ler a documentação, tem muito material interessante, mas não sei se ele te fará um desenvolvedor completo na ferramenta.

**Reply by sou-gabriel** [reply]
Boa resposta. Realmente mesclar um pouco desses 2 mundos será a melhor alternativa. Obrigado pela contribuição amigo!

**Comment by nazarepiedady** [comment]
É possível.

**Comment by MiqueiasGFernandes** [comment]
É possível sim, inclusive se pudesse voltar no tempo, hoje não gastaria mais um centavo com cursos básicos.

Mas vale ressaltar que cada um aprende de um jeito diferente, antes de tudo, descubra qual é o seu "perfil" de estudante.

Eu recomendo estudar por contra própria seguindo um roadmap, vá lendo artigos no medium, assistindo tutoriais, lendo blogs, lendo documentação e principalmente praticando BASTANTE, que em breve você chega onde deseja.

Recomendo esse roadmap: https://roadmap.sh/react
